devicesinlan.1
.TH DEVICESINLAN 1 2017-02-07
.SH NAME
devicesinlan \- Scans all devices in your LAN. Then you can set an alias to your known devices in order to detect future strange devices in your net.

.SH SYNOPSIS
.TP
.B devicesinlan [ \fB\-\-wizard\fR | \fB\-\-interface\fR | \fB\-\-add\fR | \fB\-\-remove\fR | \fB\-\-list\fR ]

.SH DESCRIPTION GUI MODE
If you launch devicesinlan without parameters and you are in a graphic system (Linux or Windows) it launches the program with a Qt Interface.

In the app menu you have the followings features:

.TP
.B Devices > New Scan

.TP
.B Devices > Show devices database

.TP
.B Devices > Reset database

This option erases all known devices in database. 

.TP
.B Configuracion > Settings

In this dialog you can select your prefered language and you can configure the number of concurrence request.

.TP
.B Help > Help

.TP
.B Help > About

.TP
.B Help > Exit


.SH DESCRIPTION CONSOLE MODE
If you launch deviceslan with some parameters, it will be executed in console mode.

.TP
.B \-\-wizard. 
It list all interfaces in the system and lets you to select the one you wish and the number of concurrent request. After that, it scans the net and prints a list of the detected devices.

.TP
.B \-\-interface
Scans the net of the interface parameter and prints a list of the detected devices.

If a device is not known, it will be showed in red. Devices in green are trusted devices.

.TP
.B \-\-add

.TP
.B \-\-remove

.TP
.B \-\-list



RELEASES.txt
devicesinlan-0.1.0 <==> svn -r   3 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.2.0 <==> svn -r   5 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.3.0 <==> svn -r   7 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.4.0 <==> svn -r  13 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.5.0 <==> svn -r  15 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.6.0 <==> svn -r  30 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.7.0 <==> svn -r  65 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.8.0 <==> svn -r  69 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.9.0 <==> svn -r  76 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.10.0 <==> svn -r 80 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.11.0 <==> svn -r 85 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-1.0.0 <==> svn -r 106 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan




devicesinlan.py
#!/usr/bin/python3
import argparse
import datetime
import platform
import sys
import signal
import logging

def signal_handler(signal, frame):
        logging.warning(Style.BRIGHT+Fore.RED+app.translate("devicesinlan","You pressed 'Ctrl+C', exiting..."))
        sys.exit(0)

#############################3

if platform.system()=="Windows":
    sys.path.append("ui")
    sys.path.append("images")
elif platform.system()=="Linux":
    sys.path.append("/usr/lib/devicesinlan")
from libdevicesinlan import ArpScanMethod, Device, Mem, SetDevices, dateversion,  version,  input_int
from colorama import init,  Style, Fore
init(autoreset=True)

if '--wizard' in sys.argv or '--interface' in sys.argv or '--add' in sys.argv or '--remove' in sys.argv or '--list' in sys.argv:
    from PyQt5.QtCore import QCoreApplication
    console=True
    app=QCoreApplication(sys.argv)
    tr=QCoreApplication.translate
else:
    from PyQt5.QtWidgets import QApplication
    console=False
    app=QApplication(sys.argv)
    tr=QApplication(sys.argv).translate

app.setOrganizationName("DevicesInLAN")
app.setOrganizationDomain("devicesinlan.sourceforge.net")
app.setApplicationName("DevicesInLAN")


signal.signal(signal.SIGINT, signal_handler)


parser=argparse.ArgumentParser(prog='devicesinlan', description=app.translate("devicesinlan",'Show devices in a LAN making an ARP and a ICMP request to find them'),  
epilog=app.translate("devicesinlan","If you like this app, please vote for it in Sourceforge (https://sourceforge.net/projects/devicesinlan/reviews/).")+"\n"
      +app.translate("devicesinlan","Developed by Mariano Muñoz 2015-{}".format(dateversion.year))
, formatter_class=argparse.RawTextHelpFormatter)
parser.add_argument('--version', action='version', version="{} ({})".format(version, dateversion))
group = parser.add_mutually_exclusive_group()
group.add_argument('--wizard', help=app.translate("devicesinlan",'Uses a wizard to select options'), action='store_true',  default=False)
group.add_argument('--interface', help=app.translate("devicesinlan",'Net interface name'))
group.add_argument('--add', help=app.translate("devicesinlan",'Add a known device'), action='store_true')
group.add_argument('--remove', help=app.translate("devicesinlan",'Remove a known device'), action='store_true')
group.add_argument('--list', help=app.translate("devicesinlan",'List known devices'), action='store_true')
parser.add_argument('--debug', help=app.translate("devicesinlan", "Debug program information"), default="WARNING")
args=parser.parse_args()        

#Por defecto se pone WARNING y mostrar´ia ERROR y CRITICAL
logFormat = "%(asctime)s %(levelname)s %(module)s:%(lineno)d at %(funcName)s. %(message)s"
dateFormat='%Y%m%d %I%M%S'

if args.debug=="DEBUG":#Show detailed information that can help with program diagnosis and troubleshooting. CODE MARKS
    logging.basicConfig(level=logging.DEBUG, format=logFormat, datefmt=dateFormat)
elif args.debug=="INFO":#Everything is running as expected without any problem. TIME BENCHMARCKS
    logging.basicConfig(level=logging.INFO, format=logFormat, datefmt=dateFormat)
elif args.debug=="WARNING":#The program continues running, but something unexpected happened, which may lead to some problem down the road. THINGS TO DO
    logging.basicConfig(level=logging.WARNING, format=logFormat, datefmt=dateFormat)
elif args.debug=="ERROR":#The program fails to perform a certain function due to a bug.  SOMETHING BAD LOGIC
    logging.basicConfig(level=logging.ERROR, format=logFormat, datefmt=dateFormat)
elif args.debug=="CRITICAL":#The program encounters a serious error and may stop running. ERRORS
    logging.basicConfig(level=logging.CRITICAL, format=logFormat, datefmt=dateFormat)
else:
    logging.basicConfig(level=logging.CRITICAL, format=logFormat, datefmt=dateFormat)
    logging.critical("--debug parameter must be DEBUG, INFO, WARNING, ERROR or CRITICAL")
    sys.exit(1)
    
mem=Mem()
mem.setApp(app)
mem.change_language(mem.settings.value("frmSettings/language", "en"))
mem.setInstallationUUID()

if console==False:    
    app.setQuitOnLastWindowClosed(True)
    import frmMain 
    frmMain = frmMain.frmMain(mem) 
    frmMain.show()
    sys.exit(app.exec_())
else:##Console
    if args.add==True:
        d=Device(mem)
        d.insert_mac()
        d.insert_alias()
        d.insert_type()
        d.link()
        print (Style.BRIGHT+ Fore.GREEN + app.translate("devicesinlan","Device inserted"))
        mem.settings.sync()
        sys.exit(0)
       
    if args.remove==True:
        d=Device(mem)
        d.insert_mac()
        d.unlink()
        print (Style.BRIGHT+Fore.GREEN+app.translate("devicesinlan","Mac removed"))

        mem.settings.sync()
        sys.exit(0)
        
    if args.list==True:
        set=SetDevices(mem)
        set.init__from_settings()
        set.print_devices_from_settings()
        sys.exit(0)
    ## Load devices
    if args.interface:
        if mem.interfaces.find_by_id(args.interface)==None:
            print(Style.BRIGHT+Fore.RED+app.translate("devicesinlan", "This interface doesn't exist. Please use --wizard parameter to help you."))
            sys.exit(1)
        mem.interfaces.selected=mem.interfaces.find_by_id(args.interface)
        
    if args.wizard==True:
        if mem.interfaces.length()==0:
            print(Style.BRIGHT+ Fore.RED+app.translate("devicesinlan", "There are not interfaces to scan."))
            sys.exit(1)
        mem.interfaces.print()
        while True:
            id=input_int(app.translate("devicesinlan", "Select an interface number"), 1)
            if id<=mem.interfaces.length():#Check id 
                break
        mem.interfaces.selected=mem.interfaces.find_by_id(mem.interfaces.arr[id-1].id)
        mem.settings.setValue("frmSettings/concurrence", input_int(app.translate("devicesinlan", "Input an integer with the request concurrence"), mem.settings.value("frmSettings/concurrence", 200)))
        
    
    inicio=datetime.datetime.now()
    set=SetDevices(mem)
    set.setMethod(ArpScanMethod.PingArp)
    set.print()
    print (Style.BRIGHT+app.translate("devicesinlan","It took {} with DevicesInLAN scanner.").format (Fore.GREEN+str(datetime.datetime.now()-inicio)+ " "+ app.translate("devicesinlan", "seconds")+Fore.WHITE))


